src/app/ipv/middleware.js

```diff
@@ -210,6 +211,40 @@ function checkJourneyAction(req) {
   }
 }
 
+// getCoiUpdateDetailsJourney determines the next journey based on the detailsToUpdate
+// field of the update-details page
+function getCoiUpdateDetailsJourney(detailsToUpdate) {
+  if (detailsToUpdate) {
+    // convert to array if its a string
+    if (typeof detailsToUpdate === "string") {
+      detailsToUpdate = [detailsToUpdate];
+    }
+    // cancel event
+    if (detailsToUpdate.includes("cancel")) {
+      return UPDATE_DETAILS_JOURNEY_TYPES.UPDATE_CANCEL;
+    }
+    // send to update-names-dob journey if dateOfBirth or both names selected
+    if (
+      detailsToUpdate.includes("dateOfBirth") ||
+      (detailsToUpdate.includes("givenNames") &&
+        detailsToUpdate.includes("lastName"))
+    ) {
+      return UPDATE_DETAILS_JOURNEY_TYPES.UPDATE_NAMES_DOB;
+    }
+    // send to update-name-address journey if only one name selected
+    if (
+      detailsToUpdate.includes("givenNames") ||
+      detailsToUpdate.includes("lastName")
+    ) {
+      return UPDATE_DETAILS_JOURNEY_TYPES.UPDATE_NAME_ADDRESS;
+    }
```

Comment:
Possibly I've gotten confused, but do we not need to know whether the user has also selected Address here? The Name Change sub journey has two separate journeys within it, one for Name + Address and the other for just name

Reply:
@RogVanGDS 
So according to the ticket:
- DOB with anything OR both names selected (plus anything else I guess) goes to UPDATE_NAME_DOB
- Either name on its own OR with an address goes to UPDATE_NAME_ADDRESS
- Address on its own goes to UPDATE_ADDRESS

Theres not much point checking for the address in this case  as its going to the same journey regardless

Reply:
Hm ok I think the ticket might actually be wrong, or something has fallen in the cracks between your ticket and [mine](https://govukverify.atlassian.net/browse/PYIC-5829).

If the user needs to change Address + Name, we delete all their VCs take them down to DCMAW CRI ->  ADDRESS CRI -> FRAUD CRI

If the user needs to just change Name (without address), we delete all their VCs **except** the Address VC, and we take them down DCMAW CRI -> FRAUD CRI.

So we will need a separate route from the front-end so the backend can differentiate between the two.

@Joe-Edwards-GDS Not sure if there is a separate ticket for this or if it's just been missed?

Reply:
@RogVanGDS so my understanding is that once the user selects to change a name and/or address in this screen, then they get dropped into the sub-journey referenced in your ticket which takes them through either the name only or name+address journeys. Maybe I'm missing something though

Reply:
I think @RogVanGDS is right - we'll want separate events for name vs name + address. I expect that they'll both go to the same sub-journey, but will go to different entry-points.

Sorry, I hadn't originally realised that we needed to skip address if the user doesn't elect to change it. If it causes a lot of difficulty we could try and push back on the UCD, but it feels doable.

(More broadly, this is starting to stretch the capability of what our journey map can do - there's a piece of work in the roadmap to evaluate what the evolution of IPV Core journey routing looks like, but I don't want to half-bake it)

Reply:
Thanks for the comments @Joe-Edwards-GDS @RogVanGDS I've added a separate event for name only, please see latest commit :)